### PR TITLE
custom role

### PR DESCRIPTION
**Pull request by mbedulli**
### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/coralogix/terraform-provider-coralogix/blob/master/CHANGELOG.md):

```release-note
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: coralogix/resource_coralogix_custom_role.go [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+			"permissions": schema.SetAttribute{
+				Required:            true,
+				ElementType:         types.StringType,
+				MarkdownDescription: "Custom role permissions",
+				Validators: []validator.Set{
+					setvalidator.SizeAtLeast(1),
+				},
+			},
+		},
+		MarkdownDescription: "Coralogix Api keys.",
```
Comment (OrNovo) [inline coralogix/resource_coralogix_custom_role.go:106]: wrong description

File: coralogix/resource_coralogix_custom_role.go [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+	}
+
+	if currentState.Name != desiredState.Name {
+		updateRoleRequest.NewName = desiredState.Name.ValueStringPointer()
+	}
+	if currentState.Description != desiredState.Description {
+		updateRoleRequest.NewDescription = desiredState.Description.ValueStringPointer()
+	}
+
+	if !reflect.DeepEqual(currentState.Permissions.Elements(), desiredState.Permissions.Elements()) {
```
Comment (OrNovo) [inline coralogix/resource_coralogix_custom_role.go:249]: use `currentState.Permissions.Equaul(desiredState.Permissions)` instead.

File: coralogix/resource_coralogix_custom_role.go [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+		}
+		return nil, diags
+	}
+
+	model, diags := flatterCustomRole(ctx, createCustomRoleResponse.GetRole())
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	model.Permissions, diags = types.SetValueFrom(ctx, types.StringType, model.Permissions.Elements())
```
Comment (OrNovo) [inline coralogix/resource_coralogix_custom_role.go:201]: What is the purpose of this line?

File: coralogix/resource_coralogix_custom_role.go [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Custom Role name.",
+			},
+			"description": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Custom Role description.",
+			},
+			"team_id": schema.Int64Attribute{
+				Required:            true,
```
Comment (OrNovo) [inline coralogix/resource_coralogix_custom_role.go:90]: Let's discuss it

File: coralogix/resource_coralogix_custom_role.go (hunk 3 above)
Comment (mbedulli) [reply coralogix/resource_coralogix_custom_role.go:201]: None some left over

**Review by OrNovo — APPROVED** [review]

**Pull request merged**